src/pow.rs

```diff
@@ -13,8 +14,11 @@ mod hasher;
 mod heavy_hash;
 mod xoshiro;
 
+static STATE_ID: AtomicUsize = AtomicUsize::new(0);
```

Comment:
Can you make the `id` a field in `MinerManager` and then pass it in to `State::new`?
Also, please make this a `u64` to make sure that it doesn't overflow even in 32bit machines

Reply:
So because the check is that `id` is not equal to last `id` I guess this can stay a `usize`

Reply:
The `id` of the state is not generated in `MinerManger` and pass to the `State`